app/plugin.go

```diff
@@ -182,6 +189,78 @@ func (a *App) InitPlugins(pluginDir, webappPluginDir string) {
 	a.SyncPluginsActiveState()
 }
 
+// SynchPlugins will synchronize the plugins installed locally
+// with the plugin bundles available in the file store.
+func (a *App) SynchPlugins() *model.AppError {
+	mlog.Info("Synching plugins with the file store ...")
+
+	pluginDirs, err := ioutil.ReadDir(*a.Config().PluginSettings.Directory)
+	if err != nil {
+		return model.NewAppError("SynchPlugins", "app.plugin.synch.read_local_folder.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	// Remove plugins locally to synch with file store.
+	for _, dir := range pluginDirs {
+		if !dir.IsDir() {
+			continue
+		}
+		dirFullPath := filepath.Join(*a.Config().PluginSettings.Directory, dir.Name())
+		files, readErr := ioutil.ReadDir(dirFullPath)
+
+		if readErr != nil {
+			mlog.Error("Error reading local plugin directoy. Skipped for synch.", mlog.String("folder", dir.Name()), mlog.Err(readErr))
+			continue
+		}
+		// Only handle managed plugins with .filestore flag file.
+		managed := false
+		for _, f := range files {
+			if f.Name() == ".filestore" {
+				managed = true
+
+				mlog.Debug("Plugin Synch: Uninstalling plugin locally", mlog.String("plugin", dir.Name()))
+				if err := a.removePluginLocally(dir.Name()); err != nil {
+					mlog.Error("Plugin Synch: Error uninstalling managed plugin.", mlog.String("plugin", dir.Name()), mlog.Err(err))
+				}
+				break
+			}
+		}
+
+		if !managed {
+			mlog.Warn("Found unmanaged plugin. Ignoring in plugin synch with the filestore.", mlog.String("plugin", dir.Name()))
+		}
+	}
+
+	// Install plugins from the file store.
+	exists, existsErr := a.FileExists("./plugins")
+
+	if existsErr != nil {
```

Comment:
nit: re: new `existsErr` instead of just using `err`

Reply:
I've renamed it to `appErr` to avoid conflicts of error type to the previously assigned `err`